tests/integration/pilot/main_test.go

```diff
@@ -90,49 +95,71 @@ values:
 				p.ServicePort = p.InstancePort
 				headlessPorts[i] = p
 			}
-			if _, err := echoboot.NewBuilder(ctx).
-				With(&apps.podA, echo.Config{
-					Service:   "a",
-					Namespace: apps.namespace,
-					Ports:     echoPorts,
-					Subsets:   []echo.SubsetConfig{{}},
-				}).
-				With(&apps.podB, echo.Config{
-					Service:   "b",
-					Namespace: apps.namespace,
-					Ports:     echoPorts,
-					Subsets:   []echo.SubsetConfig{{}},
-				}).
-				With(&apps.headless, echo.Config{
-					Service:   "headless",
-					Headless:  true,
-					Namespace: apps.namespace,
-					Ports:     headlessPorts,
-					Subsets:   []echo.SubsetConfig{{}},
-				}).
-				With(&apps.vmA, echo.Config{
-					Service:    "vm-a",
-					Namespace:  apps.namespace,
-					Ports:      echoPorts,
-					DeployAsVM: true,
-					Subsets:    []echo.SubsetConfig{{}},
-				}).
-				With(&apps.naked, echo.Config{
-					Service:   "naked",
-					Namespace: apps.namespace,
-					Ports:     echoPorts,
-					Subsets: []echo.SubsetConfig{
-						{
-							Annotations: map[echo.Annotation]*echo.AnnotationValue{
-								echo.SidecarInject: {
-									Value: strconv.FormatBool(false)},
+			builder := echoboot.NewBuilder(ctx)
+			for _, c := range ctx.Clusters() {
+				builder.
+					With(nil, echo.Config{
+						Service:   fmt.Sprintf("a-%d", c.Index()),
```

Comment:
don't we generally have the same service across clusters? just with instances distributed amongst the clusters?

Reply:
True; we would have to do a # of requests for every validation high enough that we can assert that we got a response from every cluster. We'd also need to make a request from each cluster. https://github.com/istio/istio/pull/25923 should do that.